fast64_internal/oot/scene/exporter/to_c/scene_header.py

```diff
@@ -1,208 +1,214 @@
 from .....utility import CData, indent
 from .....f3d.f3d_gbi import ScrollMethod, TextureExportSettings
 from ....oot_f3d_writer import OOTGfxFormatter
-from .scene_pathways import ootPathListToC
-from .actor import ootTransitionActorListToC, ootStartPositionListToC, ootEntranceListToC
-from .scene_commands import ootSceneCommandsToC
+from ....oot_level_classes import OOTScene, OOTLight
+from .scene_pathways import getPathData
+from .actor import getTransitionActorList, getSpawnActorList, getSpawnList
+from .scene_commands import getSceneCommandList
 
 
 ##################
 # Light Settings #
 ##################
-def ootVectorToC(vector):
-    return f"0x{vector[0]:02X}, 0x{vector[1]:02X}, 0x{vector[2]:02X}"
-
-
-def ootLightToC(light):
-    return (
-        indent
-        + "{ "
-        + ", ".join(
-            (
-                ootVectorToC(light.ambient),
-                ootVectorToC(light.diffuseDir0),
-                ootVectorToC(light.diffuse0),
-                ootVectorToC(light.diffuseDir1),
-                ootVectorToC(light.diffuse1),
-                ootVectorToC(light.fogColor),
-                light.getBlendFogShort(),
-                f"0x{light.fogFar:04X}",
-            )
+def getColorValues(vector: tuple[int, int, int]):
+    return ", ".join(f"{v:5}" for v in vector)
+
+
+def getDirectionValues(vector: tuple[int, int, int]):
+    return ", ".join(f"{v - 0x100 if v > 0x7F else v:5}" for v in vector)
+
+
+def getLightSettingsEntry(light: OOTLight, lightMode: str, isLightingCustom: bool, index: int):
+    vectors = [
+        (light.ambient, "Ambient Color", getColorValues),
+        (light.diffuseDir0, "Diffuse0 Direction", getDirectionValues),
+        (light.diffuse0, "Diffuse0 Color", getColorValues),
+        (light.diffuseDir1, "Diffuse1 Direction", getDirectionValues),
+        (light.diffuse1, "Diffuse1 Color", getColorValues),
+        (light.fogColor, "Fog Color", getColorValues),
+    ]
+
+    fogData = [
+        (light.getBlendFogNear(), "Blend Rate & Fog Near"),
+        (f"{light.fogFar}", "Fog Far"),
+    ]
+
+    lightDescs = ["Dawn", "Day", "Dusk", "Night"]
+
+    if not isLightingCustom and lightMode == "false":
+        lightDesc = f"// {lightDescs[index]} Lighting\n"
```

Comment:
what if `index >= 4` ?

Note the first four are for lightConfig 0, the next four are also dawn/day/dusk/night (but for lightConfig 1), ...

See `sTimeBasedLightConfigs` in the oot decomp for details, this pattern works for lightConfig 0, 1, 2, 3,
but lightConfig 4 uses settings 20 to 23 pepega

There's also a bunch of more or less hardcoded stuff. The switching between lightConfigs is mostly typically done by the en_weather_tag actor, see the `WeatherMode` enum